### PR TITLE
Make updates for the latest Spark 3.0 branch

### PR DESCRIPTION
**Pull request by tgravescs**
# Checklist before submitting (remove before creating the PR)

- [ ] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Part of issue #1447.  Not sure if you want to say Fixes it since Spark 3.0 isn't out yet.

The latest branch for Spark 3.x moved the location of from_arrow_type.  It also now errors out on a couple of other sql things. One is if the column name is ambiguous it now errors out, and it was errorring on the trend column, so I changed the name on that before the join.   The other error was treating Promo2SinceWeek as a double so I cast it to an int.

Note that I have not been able to run these end to end as my setup if failing on the training part. I have run ETL sides successfully.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: horovod/spark/common/util.py [hunk 1]
```diff
@@ -23,7 +23,11 @@
 from pyspark.ml.linalg import DenseVector, SparseVector, Vector, VectorUDT
 from pyspark.sql.types import ArrayType, BinaryType, BooleanType, FloatType, DoubleType, \
     IntegerType, LongType, NullType, StringType
-from pyspark.sql.types import from_arrow_type
+try:
+    from pyspark.sql.types import from_arrow_type
```
Comment (tgaddair) [inline horovod/spark/common/util.py:27]: Can you swap the order so we try Spark 3 import first?

**Comment by tgaddair** [comment]
Thanks for the PR @tgravescs.  To fix the sync issue, can you run:

```
diff examples/keras_spark_rossmann_run.py examples/keras_spark3_rossmann.py > examples/keras_spark3_rossmann.py.patch
```

**Comment by tgaddair** [comment]
Thanks, last thing is to sign the DCO for each commit: https://github.com/horovod/horovod/pull/1967/checks?check_run_id=693446249

**Comment by EnricoMi** [comment]
The `pytorch_spark_mnist.py` example is also broken for pyspark 3.0.0: https://github.com/horovod/horovod/pull/1957

**Comment by liangz1** [comment]
Hi @tgaddair, it seems that this PR is not included in v0.19.5. Do you know what is the plan to release this patch? Thanks!

**Pull request merged**